Content.Client/DragDrop/DragDropSystem.cs

```diff
@@ -18,479 +16,525 @@
-            {
-                return false;
-            }
+            return false;
+        }
 
-            // cancel any current dragging if there is one (shouldn't be because they would've had to have lifted
-            // the mouse, canceling the drag, but just being cautious)
-            _dragDropHelper.EndDrag();
+        var ev = new CanDragEvent();
 
-            // possibly initiating a drag
-            // check if the clicked entity is draggable
-            if (!EntityManager.EntityExists(args.EntityUid))
-            {
-                return false;
-            }
+        RaiseLocalEvent(args.EntityUid, ref ev);
 
-            // check if the entity is reachable
-            if (!_interactionSystem.InRangeUnobstructed(dragger, args.EntityUid))
-            {
-                return false;
-            }
+        if (ev.Handled != true)
+            return false;
 
-            var canDrag = false;
-            foreach (var draggable in EntityManager.GetComponents<IDraggable>(args.EntityUid))
-            {
-                var dragEventArgs = new StartDragDropEvent(dragger, args.EntityUid);
+        _draggedEntity = args.EntityUid;
+        _state = DragState.MouseDown;
+        _mouseDownScreenPos = _inputManager.MouseScreenPosition;
+        _mouseDownTime = 0;
 
-                if (!draggable.CanStartDrag(dragEventArgs))
-                {
-                    continue;
-                }
+        // don't want anything else to process the click,
+        // but we will save the event so we can "re-play" it if this drag does
+        // not turn into an actual drag so the click can be handled normally
+        _savedMouseDown = args;
 
-                _draggables.Add(draggable);
-                canDrag = true;
-            }
+        return true;
+
+    }
+
+    private void StartDrag()
+    {
+        if (!Exists(_draggedEntity))
+        {
+            // something happened to the clicked entity or we moved the mouse off the target so
+            // we shouldn't replay the original click
+            return;
+        }
+
+        _state = DragState.Dragging;
+        DebugTools.Assert(_dragShadow == null);
+        _outline.SetEnabled(false);
+        HighlightTargets();
 
-            if (!canDrag)
+        if (TryComp<SpriteComponent>(_draggedEntity, out var draggedSprite))
+        {
+            // pop up drag shadow under mouse
+            var mousePos = _eyeManager.ScreenToMap(_inputManager.MouseScreenPosition);
+            _dragShadow = EntityManager.SpawnEntity("dragshadow", mousePos);
+            var dragSprite = Comp<SpriteComponent>(_dragShadow.Value);
+            dragSprite.CopyFrom(draggedSprite);
+            dragSprite.RenderOrder = EntityManager.CurrentTick.Value;
+            dragSprite.Color = dragSprite.Color.WithAlpha(0.7f);
+            // keep it on top of everything
+            dragSprite.DrawDepth = (int) DrawDepth.Overlays;
+            if (!dragSprite.NoRotation)
```

Comment:
I forgot how cursed this was. This means you'll be able to see the shadow entity floating around on camera monitors...

This needs a  //TODO & a github issue, it should really be using an overlay & using space-wizards/RobustToolbox/pull/3566 or whatever ends up replacing it.